x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_alerts_query.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback, useMemo, useState } from 'react';
+import createContainer from 'constate';
+import { getTime } from '@kbn/data-plugin/common';
+import { TIMESTAMP } from '@kbn/rule-data-utils';
+import { buildEsQuery, Query } from '@kbn/es-query';
+import { ALERT_STATUS_QUERY } from '@kbn/observability-plugin/public';
+import { AlertStatus } from '@kbn/observability-plugin/common';
+import { SnapshotNode } from '../../../../../common/http_api';
+import { useUnifiedSearchContext } from './use_unified_search';
+import { HostsState } from './use_unified_search_url_state';
+import { useHostsView } from './use_hosts_view';
+
+export const useAlertsQueryImpl = () => {
+  const { hostNodes } = useHostsView();
+
+  const { unifiedSearchDateRange } = useUnifiedSearchContext();
+
+  const [alertStatus, setAlertStatus] = useState<AlertStatus>('all');
+
+  const getAlertsEsQuery = useCallback(
+    (status?: AlertStatus) =>
+      createAlertsEsQuery({ dateRange: unifiedSearchDateRange, hostNodes, status }),
+    [hostNodes, unifiedSearchDateRange]
+  );
+
+  const alertsEsQuery = useMemo(() => getAlertsEsQuery(), [getAlertsEsQuery]);
+
+  const alertsEsQueryByStatus = useMemo(
+    () => getAlertsEsQuery(alertStatus),
+    [getAlertsEsQuery, alertStatus]
+  );
```

Comment:
**note**: generate two different queries, `alertsEsQuery` always fetches the count of any alert, `alertsEsQueryByStatus` fetched the data filtered by status.
This is necessary because the summary and table components can be filtered with the ButtonGroup, while the TabBadge count should always return the full active alerts count.